scripts/attack_navigator.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import json
+
+import requests
+
+'''
+Use this script to generate a layer for attack-navigator
+https://github.com/mitre-attack/attack-navigator
+'''
+
+
+def get_abilities(url):
+    payload = json.dumps(dict(index='ability'))
+    headers = {
+        'Content-Type': 'application/json',
+        'API_KEY': 'ADMIN123',
+    }
+    response = requests.request('POST', url=url + '/plugin/chain/full', data=payload, headers=headers)
+    return json.loads(response.text)
+
+
+def get_layer_boilerplate(name, description):
+    return {
+        'version': '2.2',
+        'name': name,
+        'description': description,
+        'domain': 'mitre-enterprise',
+        'techniques': [],
+        'legendItems': [],
+        'showTacticRowBackground': True,
+        'tacticRowBackground': '#205b8f',
+        'selectTechniquesAcrossTactics': True,
+        'gradient': {
+            'colors': [
+                '#ffffff',
+                '#66ff66'
+            ],
+            'minValue': 0,
+            'maxValue': 1
+        }
+    }
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-o', '--out-file', default='stockpile-techniques.json', help='layer json output path name')
+    parser.add_argument('--url', default='http://127.0.0.1:8888', help='url if not default')
+    args = parser.parse_args()
+
+    layer = get_layer_boilerplate(name='stockpile', description='stockpile')
+    abilities = get_abilities(url=args.url)
+
+    for ability in abilities:
+        technique = {
+            'techniqueID': ability['technique_id'],
+            'score': 1,
+            'color': '',
+            'comment': '',
+            'enabled': True
+        }
+
+        layer['techniques'].append(technique)
+
+    with open(args.out_file, 'w') as f:
+        f.write(json.dumps(layer, indent=4))
+
+    print('now navigate to http://localhost:4200/ and upload the newly created output file')
```

Comment:
4200?

Reply:
this would require https://github.com/mitre-attack/attack-navigator to be installed, but I'm looking at https://mitre-attack.github.io/attack-navigator/enterprise/ and seeing if it can just load from there and download

Reply:
nice

Reply:
doesn't exist and I would have to implement something and do a PR for attack-navigator there is an open issue for it and Isabel says it's in the dev pipeline:
https://github.com/mitre-attack/attack-scripts/issues/1

Reply:
lmk how you'd like to incorporate this

Reply:
I think this is really good-- and could be a really cool addition to the stockpile. 

it is more work, but I think we need to automate the connection between the stockpile plugin GUI (none currently) and the https://mitre-attack.github.io/attack-navigator/enterprise/ website. 

lots of ways we could do this, but a quick one may be to simply open a new tab when you click 'stockpile' from caldera and load the output JSON into it (instead of writing to a file)

Reply:
good idea, I've moved to WIP until it's updated